homeassistant/components/linky/sensor.py

```diff
@@ -114,6 +105,12 @@ def __init__(self, name, account: LinkyAccount, scale, when):
         self._username = account.username
         self._time = None
         self._consumption = None
+        self._unique_id = f"{DOMAIN}_{self._username}_{scale}_{when}"
```

Comment:
The domain should not be part of the unique_id. The entity registry is aware of the domain and platform of each entry.